release-content/0.15/release-notes/15800_Add_mesh_picking_backend_and_MeshRayCast_system_parameter.md

```diff
@@ -1,4 +1,117 @@
+
+First, you might want to quickly update the state of your objects (be they UI or game objects) based on what is being done to them, typically highlighting them or changing their color. For that, simply match against the [`PickingInteraction`] component.
+
+Secondly, you might want to respond dynamically to various pointer-powered events. For that, we recommend using observers (which replaced the existing `bevy_event_listener` solution during the upstreaming process).
+Here, we're spawning a simple text node and responding to pointer events.
+
+```rust
+fn main() {
+    use bevy::prelude::*;
+
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, update_counter_display)
+        // Observers added globally will watch for events sent to *any* entity
+        .add_observer(change_text_color_on_hover)
+        .add_observer(reset_text_color)
+        .run();
+}
+
+fn change_text_color_on_hover(
+    over: Trigger<Pointer<Over>>,
+    mut text_colors: Query<&mut TextColor>,
+) {
+    if let Ok(mut text_color) = text_colors.get_mut(over.entity()) {
+        *text_color = Srgba::RED.into();
+    }
+}
+fn reset_text_color(out: Trigger<Pointer<Out>>, mut text_colors: Query<&mut TextColor>) {
+    if let Ok(mut text_color) = text_colors.get_mut(out.entity()) {
+        *text_color = TextColor::default();
+    }
+}
+#[derive(Component)]
+struct Counter(i32);
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2d::default());
+    // Root node
+    commands
+        .spawn(Node {
+            flex_direction: FlexDirection::Column,
+            ..Default::default()
+        })
+        .with_children(|builder| {
+            builder.spawn((Text::new("Counter: 0"), Counter(0)));
+            builder
+                .spawn((Text::new("Count up :)"),))
+                // Observers added to a single entity only watch for events to that specific entity
+                .observe(
+                    |_click: Trigger<Pointer<Click>>,
+                     mut counter_query: Query<&mut Counter>| {
+                        if let Ok(mut counter) = counter_query.get_single_mut() {
+                            counter.0 += 1;
+                        }
+                    },
+                );
+            builder.spawn((Text::new("Count down :("),)).observe(
+                |_click: Trigger<Pointer<Click>>, mut counter_query: Query<&mut Counter>| {
+                    if let Ok(mut counter) = counter_query.get_single_mut() {
+                        counter.0 -= 1;
+                    }
+                },
+            );
+        });
+}
+// Systems with the new `Single` system param are skipped if their query doesn't return exactly one elemnent
+fn update_counter_display(singleton_query: Single<(&mut Text, &Counter)>) {
+    let (mut text, counter) = singleton_query.into_inner();
+    *text = Text::new(format!("Counter: {}", counter.0));
+}
+```
```

Comment:
I think this example is a bit lengthy without any explanation in-between. Even though it glosses over some details, I think breaking it up to just focus on hovering and clicking separately can help:

_I wanted to post the below as a suggestion but GitHub doesn't like the multiple code blocks inside a suggestion block._

For example, you can use `Pointer<Over>` and `Pointer<Out>` to highlight entities under the pointer.

```rust
fn change_text_color_on_hover(
    over: Trigger<Pointer<Over>>,
    mut text_colors: Query<&mut TextColor>,
) {
    if let Ok(mut text_color) = text_colors.get_mut(over.entity()) {
        *text_color = Srgba::RED.into();
    }
}

fn reset_text_color(
    out: Trigger<Pointer<Out>>,
    mut text_colors: Query<&mut TextColor>,
) {
    if let Ok(mut text_color) = text_colors.get_mut(out.entity()) {
        *text_color = TextColor::default();
    }
}
```

You could also `Pointer<Click>` to let users interact with entities.

```rust
#[derive(Component)]
struct Counter(i32);

fn setup(mut commands: Commands) {
    commands
        .spawn(Node {
            flex_direction: FlexDirection::Column,
            ..Default::default()
        })
        .with_children(|builder| {
            builder.spawn((Text::new("Counter: 0"), Counter(0)));
            builder
                .spawn((Text::new("Count up :)"),))
                // Observers added to a single entity only watch for events to that specific entity
                .observe(
                    |_click: Trigger<Pointer<Click>>,
                     mut counter_query: Query<&mut Counter>| {
                        if let Ok(mut counter) = counter_query.get_single_mut() {
                            counter.0 += 1;
                        }
                    },
                );
        });
}
```